lib/launchers/openstack10.rb

```diff
@@ -687,6 +687,29 @@ def assign_ip(instance)
       end
     end
 
+    def allocate_floating_ip(network_name)
+      request_url = self.os_network_url + "/v2.0/floatingips"
+      method = "POST"
+      network = get_networks.find {|n| n["name"] == network_name}
+      unless network
+        raise "could not find network in current tenant."
+      end
+      network_id = network["id"]
+      payload = {
+        floatingip: {
+          floating_network_id: network_id,
+          project_id: self.os_tenant_id,
+          description: "IP allocated automatically by BushSlicer"
+        }
+      }
+      res = self.rest_run(request_url, method, payload, self.os_token)
+      if res[:success]
+        return res
```

Comment:
Returning whole result is inconvenient for user. Better return `res.dig(:parsed, "floatingip")` such that user gets the information about the IP without noise.

Reply:
got you